billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/AwsStorageToBillingRequestConverter.java

```diff
@@ -0,0 +1,297 @@
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.YearMonth;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Slf4j
+@SuppressWarnings("checkstyle:MagicNumber")
+public class AwsStorageToBillingRequestConverter implements EntityToBillingRequestConverter<AbstractDataStorage> {
+
+    private static final String STORAGE_SIZE_AGG_NAME = "sizeSumSearch";
+    private static final String SIZE_FIELD = "size";
+    private static final String REGION_FIELD = "storage_region";
+    private static final int BYTES_TO_GB = 1 << 30;
+    private static final int CENTS_IN_DOLLAR = 100;
+    private static final int PRECISION = 5;
+    private static final String ES_FILE_INDEX_PATTERN = "cp-%s-file-%d";
+    private static final String INDEX_PATTERN = "%s-daily-%d-%s";
+    private static final RoundingMode ROUNDING_MODE = RoundingMode.CEILING;
+
+    private final EntityMapper<StorageBillingInfo> mapper;
+    private final ElasticsearchServiceClient elasticsearchService;
+    private final Map<Regions, StoragePricing> storagePriceListGb = new HashMap<>();
+    private BigDecimal defaultPriceGb;
+    private final StorageType storageType;
+
+    public AwsStorageToBillingRequestConverter(final EntityMapper<StorageBillingInfo> mapper,
+                                               final ElasticsearchServiceClient elasticsearchService,
+                                               final String awsStorageServiceName,
+                                               final StorageType storageType) {
+        this.mapper = mapper;
+        this.elasticsearchService = elasticsearchService;
+        this.storageType = storageType;
+        initPrices(awsStorageServiceName);
+    }
+
+    public AwsStorageToBillingRequestConverter(final EntityMapper<StorageBillingInfo> mapper,
+                                               final ElasticsearchServiceClient elasticsearchService,
+                                               final StorageType storageType,
+                                               final Map<Regions, StoragePricing> priceList) {
+        this.mapper = mapper;
+        this.elasticsearchService = elasticsearchService;
+        this.storageType = storageType;
+        this.storagePriceListGb.putAll(priceList);
+        this.defaultPriceGb = calculateDefaultPriceGb();
+    }
+
+    @Override
+    public List<DocWriteRequest> convertEntityToRequests(final EntityContainer<AbstractDataStorage> storageContainer,
+                                                         final String indexName,
+                                                         final LocalDateTime previousSync,
+                                                         final LocalDateTime syncStart) {
+        final SearchRequest searchRequest = new SearchRequest();
```

Comment:
Extract code for getting storage size into a separate method

Reply:
Done, split `convertEntityToRequests` into smaller methods